warehouse/query-core/src/main/java/datawave/query/postprocessing/tf/TermOffsetFunction.java

```diff
@@ -1,35 +1,42 @@
 package datawave.query.postprocessing.tf;
 
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
-
-import com.google.common.collect.Multimap;
 import datawave.query.attributes.Attribute;
 import datawave.query.attributes.Attributes;
 import datawave.query.attributes.Document;
 import datawave.query.attributes.DocumentKey;
 import datawave.query.util.Tuple2;
 import datawave.query.util.Tuple3;
 import datawave.query.util.Tuples;
-
 import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.hadoop.io.Text;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
 
 public class TermOffsetFunction implements com.google.common.base.Function<Tuple2<Key,Document>,Tuple3<Key,Document,Map<String,Object>>> {
     
+    // Runs the iterator to collect term frequency offsets
     private TermOffsetPopulator tfPopulator;
-    private Set<String> tfIndexOnlyFields;
     
-    public TermOffsetFunction(TermOffsetPopulator tfPopulator, Set<String> tfIndexOnlyFields) {
+    // Responsible for reducing the term frequency search space
+    private TermFrequencyHitFunction hitFunction;
+    
+    public TermOffsetFunction(TermOffsetPopulator tfPopulator, ASTJexlScript script, SortedKeyValueIterator<Key,Value> source, boolean isTld) {
         this.tfPopulator = tfPopulator;
-        this.tfIndexOnlyFields = tfIndexOnlyFields;
+        this.hitFunction = new TermFrequencyHitFunction(script, tfPopulator.getTermFrequencyFieldValues());
```

Comment:
I'm still getting my head around this, but when I see the `TermFrequencyHitFunction` embedded within `TermOffsetFunction` - I wonder if we could make them 'siblings' instead of 'parent/child' functions; Meaning: is there an advantage if the caller of `TermOffsetFunction` also called `TermFrequencyHitFunction` and passes the data from one to another in the appropriate sequence. Does this make either easier to unit test?

Reply:
It looks like there's a layer of abstraction here with `TFFactory` and we effectively have the consumer calling a specifically functional interface, so disregard my original comment. However, there's something about the relationship between `TermOffsetFunction`, `TermOffsetPopulator` and `TermFrequencyHitFunction` that seems a little asymmetrical. Perhaps it is as simple as constructing the `TermFrequencyHitFunction` in the `TFFactory` implementation the same way we do the `TermOffsetPopulator`. The other thought would be to restructure these three classes slightly so we essentially have a `TermOffsetFunction` implementation that does it all.